tpu-client/src/quic_client.rs

```diff
@@ -56,6 +104,24 @@ impl QuicTpuConnection {
     }
 }
 
+async fn send_wire_transaction_async(
+    connection: Arc<NonblockingQuicTpuConnection>,
+    wire_transaction: Vec<u8>,
+) -> TransportResult<()> {
+    let result = connection.send_wire_transaction(wire_transaction).await;
+    ASYNC_TASK_SEMAPHORE.release();
+    result
+}
+
+async fn send_wire_transaction_batch_async(
+    connection: Arc<NonblockingQuicTpuConnection>,
+    buffers: Vec<Vec<u8>>,
+) -> TransportResult<()> {
+    let result = connection.send_wire_transaction_batch(&buffers).await;
+    ASYNC_TASK_SEMAPHORE.release();
```

Comment:
@lijunwangs why the semaphore is released here? as opposed to where it was acquired?
the task spawned to run this function is not finished yet to release the semaphore.
on the other hand if we want to release it here, why not acquire it inside this function as well?